src/Engine/Utils.java

```diff
@@ -113,20 +113,25 @@ public static ArrayList<XYCoord> findPossibleDestinations(Unit unit, GameMap gam
   }
   /**
    * Finds the area reachable this turn by input unit.
-   * Alias for {@link #findFloodFillArea(XYCoord, FloodFillFunctor, int, GameMap) findFloodFillArea()}
+   * Alias for {@link #findFloodFillArea(XYCoord, MoveType, int, GameMap, boolean) findFloodFillArea()}
    */
   public static ArrayList<XYCoord> findPossibleDestinations(XYCoord start, Unit unit, GameMap gameMap, boolean includeOccupiedSpaces)
   {
-    return findFloodFillArea(start, unit.getMoveFunctor(includeOccupiedSpaces), Math.min(unit.getMovePower(gameMap), unit.fuel), gameMap);
+    boolean canTravelThroughEnemies = false;
+    return findFloodFillArea(start, unit, unit.CO.army, unit.getMoveFunctor(), Math.min(unit.getMovePower(gameMap), unit.fuel), gameMap, includeOccupiedSpaces, canTravelThroughEnemies);
   }
   /**
    * @param start Initial location; will usually be in the output set.
-   * @param fff Determines the cost to fill from one tile to another.
-   * @param initialFillPower How much juice to give the FloodFillFunctor
+   * @param mover Identity of the unit that's moving; may be null
+   * @param team Affiliation of the unit that's moving; may be null
+   * @param fff Determines the cost to move from one tile to another.
+   * @param initialFillPower How much move juice we start with
    * @param gameMap The map to search over.
+   * @param includeOccupiedSpaces Whether to include places that have units in them already; generally true if canTravelThroughEnemies is
+   * @param canTravelThroughEnemies Whether to allow pathing through enemies
    * @return The list of XYCoords in gameMap reachable by the FloodFillFunctor from the start coord, given initialFillPower.
    */
-  public static ArrayList<XYCoord> findFloodFillArea(XYCoord start, FloodFillFunctor fff, int initialFillPower, GameMap gameMap)
+  public static ArrayList<XYCoord> findFloodFillArea(XYCoord start, Unit mover, Army team, MoveType fff, int initialFillPower, GameMap gameMap, boolean includeOccupiedSpaces, boolean canTravelThroughEnemies)
```

Comment:
Why accept a `Unit`, its `Army`, and its `MoveType` all as separate arguments? If you have a valid Unit, it should already have the `Army` and the 'MoveType`.

Reply:
It's providing different levels of specificity.
The Unit is there for identity purposes: for checking whether it can stand the tile it currently occupies.
The Army is necessary for logic around pathing through other units, and can also work with null (enemy of everyone, as usual).
The MoveType is necessary to do anything; it can't be null.

Thinking about it, we don't need the Unit param if we remove `includeOccupiedSpaces`.
We would make that always true (units' tiles are put into the output set)., so we'd have to adapt some AI functions. Maybe it's a change worth making.

Reply:
Fair enough I guess.